remove an unwanted loop

jdannberg opened the pull request:
I think we do not need this loop (as we already do the loop over all quadrature points within the individual averaging functions), and in the case of the averaging schemes that do not generate one average for all of the points in one cell, but calculate position-dependent, different values for every point, this even generates different results (because we average several times). 
Please correct me if I am wrong!

tjhei commented on source/material_model/averaging.cc:
```diff
@@ -321,17 +321,14 @@ namespace aspect
        */
       if (out.viscosities.size() > 1)
```
@bangerth can you explain this logic to me? The only constructor of ```MaterialModelOutputs``` gets the number of quadrature points, so checking for 1 doesn't make sense to me.

bangerth replied:
The best I can think of is that if you have only one value, averaging it with itself will not yield anything that's terribly useful or different -- it would be a waste of time. Of course, the comment above the code is terribly unhelpful in all ways I can think of. 

If that passes all tests, I'd be ok with simply removing the `if` statement (or, if we'd like to skip the averaging, with updating the comment).

tjhei commented on source/material_model/averaging.cc:
```diff
@@ -321,17 +321,14 @@ namespace aspect
+          average (averaging_operation,in.position,out.viscosities);
+          average (averaging_operation,in.position,out.densities);
+          average (averaging_operation,in.position,out.thermal_expansion_coefficients);
+          average (averaging_operation, in.position,out.specific_heat);
```
can you please fix the spacing of the arguments, @jdannberg ? :-)

tjhei commented:
And yes, that looks much better than before.

bangerth commented:
This looks good to merge if you address the comments above (and you'll also need to rebase to master to avoid the conflicts).

jdannberg commented:
It also changes some of the test results (I think the Normalized Weighed Distance ones, because they are done only once now instead of several times), so I wanted to wait with my changes until the tester runs again.

bangerth commented:
Oh, you mean that we accidentally smoothed multiple times due to the loop? That's awkward. Can you adjust the test results too, then?

jdannberg commented:
Yes, sure. I just wanted to wait for the tester to be sure I do not forget any tests.

tjhei commented:
Okay, can you please fix the results and rebase to master?

jdannberg commented:
I updated the tests and rebased, but I don't think we can remove the if() statement. 
I tried, and it gives me a floating point exception in AdiabaticConditions/initial_profile/initialize (and the solver crashes in release mode). I think the problem is that we call the averaging scheme with only one point in that place, and then the max_distance variable in the nwd averaging schemes is zero (because we only have one point), and we divide by this variable. 
@bangerth, is this correct? Should I update the comment and leave the if-statement as is?

tjhei commented:
Makes sense. We could move the if inside the ```average()``` function though.

Please add a comment and fix what the tester is complaining about.

jdannberg commented:
I updated the comment, and I guess I'll just wait for your other two pull requests and then rebase again (I think they will fix what the tester is complaining about).

jdannberg commented:
Okay, all the tests seem to work now.
